### PR TITLE
Add AnswersController#destroy

### PR DESCRIPTION
**Pull request by OmegaLord**
Аутентифицированный пользователь может удалить свой ответ

### CODE REVIEW

File: spec/controllers/answers_controller_spec.rb [hunk 1]
```diff
@@ -96,4 +96,34 @@
     end
   end
   # rubocop:enable RSpec/NestedGroups
+
+  describe 'DELETE #destroy' do
+    before do
+      sign_in_user(answer.user)
+      answer
```
Comment (AZatsepa) [inline spec/controllers/answers_controller_spec.rb]: The answer was created in a line above when you called `answer.user`

File: spec/controllers/answers_controller_spec.rb [hunk 2]
```diff
@@ -96,4 +96,34 @@
...
+        expect(response).to render_template :destroy
+      end
+    end
+
+    describe 'not belongs to current user' do
+      it 'redirect to the index template' do
+        expect do
+          delete :destroy, params: { id: answer, question_id: answer.question.id }, format: :js
+        end
+          .not_to change(Question, :count)
```
Comment (AZatsepa) [inline spec/controllers/answers_controller_spec.rb]: `it 'redirect to the index template' do` != `.not_to change(Question, :count)`

File: spec/features/user_delete_answer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        end
+      end
+
+      it 'has delete current answer' do
+        within '.answers' do
+          delete_answer
+          expect(page).not_to have_content answer.body
+        end
+      end
+
```
Comment (AZatsepa) [inline spec/features/user_delete_answer_spec.rb]: You can check if the answer has been deleted.

Comment (OmegaLord) [reply spec/features/user_delete_answer_spec.rb]: That is the check. I think, the check if the answer has been deleted, held in controller test.

Comment (AZatsepa) [reply spec/features/user_delete_answer_spec.rb]: This check checks the only page content.

**Pull request merged**